app/routes.py

```diff
@@ -0,0 +1,345 @@
+    db.session.commit()
+
+    return {
+        "customer_id": rental.customer_id,
+        "video_id": rental.video_id,
+        "videos_checked_out_count": video_checked_out_count,
+        "available_inventory": available_inventory 
+        }, 200
+
+@customers_bp.route("/<customer_id>/rentals", methods=["GET"])
+def list_customer_rentals(customer_id):
+    customer = Customer.query.get(customer_id)
+    if customer is None:
+        return jsonify({
+            "message" : f"Customer {customer_id} was not found"
+        }), 404
+    customer_rentals = Rental.query.filter_by(customer_id=customer_id, checked_in_status=False).all()
+    customer_rentals_response = []
+    for customer_rental in customer_rentals:
+        video = Video.query.get(customer_rental.video_id)
+        customer_rentals_response.append({
+            "release_date" : video.release_date,
+            "title" : video.title,
+            "due_date" : customer_rental.due_date
+        })
+    return jsonify(customer_rentals_response)
+
+
+@videos_bp.route("/<video_id>/rentals", methods=["GET"])
+def list_video_rentals(video_id):
+    video = Video.query.get(video_id)
+    if video is None:
+        return jsonify({
+            "message" : f"Video {video_id} was not found"
+        }), 404
+    video_rentals = Rental.query.filter_by(video_id=video_id, checked_in_status=False).all()
+    video_rentals_response = []
+    for video_rental in video_rentals:
+        customer = Customer.query.get(video_rental.customer_id)
+        video_rentals_response.append({
+            "due_date" : video_rental.due_date,
+            "name" : customer.name,
+            "phone" : customer.phone,
+            "postal_code" : customer.postal_code
+        })
+    return jsonify(video_rentals_response)
+
+
+@videos_bp.route("", methods=["GET", "POST"])
+def handle_videos():
+    if request.method == "GET":
+        videos = Video.query.all()
+        videos_response = []
+        for video in videos:
+            videos_response.append({
+                "id": video.id,
+                "title": video.title,
+                "release_date": video.release_date,
+                "total_inventory": video.total_inventory
+            })
+        
+        return jsonify(videos_response), 200
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+        title = request_body.get("title")
+        release_date = request_body.get("release_date")
+        total_inventory = request_body.get("total_inventory")
+
+        if not title:
+            return jsonify({
+                "details" : "Request body must include title."
+            }), 400
+        
+        if not release_date:
+            return jsonify({
+                "details" : "Request body must include release_date."
+            }), 400
+        
+        if not total_inventory:
+            return jsonify({
+                "details" : "Request body must include total_inventory."
+            }), 400
```

Comment:
When things repeat three times, that's a great chance to look at how to dry up our code. Notice that for all three of these cases, we check for a value, and then build a dictionary with an error if it's empty. What if we built a a list of pairs of values: the value to check, and the string to use in the error, and looped over the list?